core/src/main/java/com/datastax/oss/driver/internal/core/metadata/schema/DefaultAggregateMetadata.java

```diff
@@ -40,7 +41,7 @@
   @NonNull private final DataType returnType;
   @NonNull private final FunctionSignature stateFuncSignature;
   @NonNull private final DataType stateType;
-  @NonNull private final TypeCodec<Object> stateTypeCodec;
+  @NonNull private final transient TypeCodec<Object> stateTypeCodec;
```

Comment:
Perhaps the most significant change in an otherwise straightforward PR.  It didn't seem reasonable to make the codec serializable so I went with this approach right now.  The IT still passes even though the codec won't be populated after deserialization becuase stateTypeCodec isn't included in the equals() impl.

My working assumption was that this change wouldn't be terribly significant in practice.  A missing codec here will cause AggregateMetadata.describe() will misbehave on deserialized instances.  But the presumed motivation for making these classes serializable is (at least largely) to use them directly as DTOs... and their performance in that role should still be intact here.

Perhaps another not worth documenting... but not much more?

Reply:
Correction: AggregateMetadata.describe() will be fine in this case.  It will make an attempt to use the codec but if that results in an exception (which it will in this case by way of an NPE) it falls back to toString() instead.  Still seems worthwhile adding an explicit test for this functionality as a guard against future breakage.